src/multi.c

```diff
@@ -380,14 +380,13 @@ void touchAllWatchedKeysInDb(redisDb *emptied, redisDb *replaced_with) {
     dictIterator *di = dictGetSafeIterator(emptied->watched_keys);
     while((de = dictNext(di)) != NULL) {
         robj *key = dictGetKey(de);
-        list *clients = dictGetVal(de);
-        if (!clients) continue;
-        listRewind(clients,&li);
-        while((ln = listNext(&li))) {
-            client *c = listNodeValue(ln);
-            if (dictFind(emptied->dict, key->ptr)) {
-                c->flags |= CLIENT_DIRTY_CAS;
-            } else if (replaced_with && dictFind(replaced_with->dict, key->ptr)) {
+        if (dictFind(emptied->dict, key->ptr) ||
+            (replaced_with && dictFind(replaced_with->dict, key->ptr))) {
```

Comment:
on multi-line ifs, we usually move the braces to a line of their own (looks a bit better)
```suggestion
            (replaced_with && dictFind(replaced_with->dict, key->ptr)))
            {
```